lib/model/AttributeGroup.dart

```diff
@@ -31,20 +31,20 @@ class AttributeGroup extends JsonObject {
   static const String _JSON_TAG_WARNING = 'warning';
   static const String _JSON_TAG_ATTRIBUTES = 'attributes';
 
-  final String id;
-  final String name;
-  final String warning;
-  final List<Attribute> attributes;
+  final String? id;
+  final String? name;
+  final String? warning;
+  final List<Attribute>? attributes;
 
   @override
   String toString() => 'AttributeGroup(${toJson()})';
 
-  List<Map<String, dynamic>> _listToJson() {
-    if (attributes == null || attributes.isEmpty) {
+  List<Map<String, dynamic>>? _listToJson() {
+    if (attributes == null || attributes!.isEmpty) {
```

Comment:
Just curious, what is the meaning of the !. operator?

Reply:
The ! can be seen as a force push.
 When a function accepts only not nullable but you have to use a nullable you can still parse it with the !

Reply:
Yeah so `attributes.isEmpty` would cause an error because `attributes` could potentially be null. However, in this case we can be certain that it is not so using `!` is fine.

Reply:
Of course the game is to almost never use `!`, and use non-nullable fields/parameters instead. Whenever possible.